mypyc/test-data/run-lists.test

```diff
@@ -338,3 +338,16 @@ assert list_in_mixed(0.0)
 assert not list_in_mixed([1])
 assert not list_in_mixed(object)
 assert list_in_mixed(type)
+
+[case testListBuiltFromGenerator]
+def test() -> None:
+    source = ["a", "b", "c"]
+    a = list(x + "f2" for x in source)
+    b = [x + "f2" for x in source]
```

Comment:
Make all these lists distinct (e.g. replace `f2` with `f3` here), for a bit more testing coverage.